homeassistant/components/linky/config_flow.py

```diff
@@ -0,0 +1,122 @@
+            user_input = {}
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_USERNAME, default=user_input.get(CONF_USERNAME, "")
+                    ): str,
+                    vol.Required(
+                        CONF_PASSWORD, default=user_input.get(CONF_PASSWORD, "")
+                    ): str,
+                }
+            ),
+            errors=errors or {},
+        )
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        errors = {}
+
+        if user_input is None:
+            return self._show_setup_form(user_input, None)
+
+        self._username = user_input[CONF_USERNAME]
+        self._password = user_input[CONF_PASSWORD]
+        self._timeout = user_input.get(CONF_TIMEOUT, DEFAULT_TIMEOUT)
+
+        if self._configuration_exists(self._username):
+            errors[CONF_USERNAME] = "username_exists"
+            return self._show_setup_form(user_input, errors)
+
+        client = LinkyClient(self._username, self._password, None, self._timeout)
+        try:
+            await self.hass.async_add_executor_job(client.login)
+            await self.hass.async_add_executor_job(client.fetch_data)
+        except PyLinkyAccessException as exp:
+            _LOGGER.error(exp)
+            errors["base"] = "access"
+            return self._show_setup_form(user_input, errors)
+        except PyLinkyEnedisException as exp:
+            _LOGGER.error(exp)
+            errors["base"] = "enedis"
+            return self._show_setup_form(user_input, errors)
+        except PyLinkyWrongLoginException as exp:
+            _LOGGER.error(exp)
+            errors["base"] = "wrong_login"
+            return self._show_setup_form(user_input, errors)
+        except PyLinkyException as exp:
+            _LOGGER.error(exp)
+            errors["base"] = "unknown"
+            return self._show_setup_form(user_input, errors)
+        finally:
+            client.close_session()
+
+        return self.async_create_entry(
+            title=self._username,
+            data={
+                CONF_USERNAME: self._username,
+                CONF_PASSWORD: self._password,
+                CONF_TIMEOUT: self._timeout,
+            },
+        )
+
+    async def async_step_import(self, user_input=None):
+        """Import a config entry.
+
+        Only host was required in the yaml file all other fields are optional
+        """
+        username = user_input[CONF_USERNAME]
+        password = user_input[CONF_PASSWORD]
+        timeout = user_input.get(CONF_TIMEOUT, DEFAULT_TIMEOUT)
+        if self._configuration_exists(username):
+            return self.async_abort(reason="username_exists")
+        return await self.async_step_user(
+            {CONF_USERNAME: username, CONF_PASSWORD: password, CONF_TIMEOUT: timeout}
```

Comment:
We don't need to create a new dict here. We can just pass on `user_input` as is.